src/middlewared/middlewared/etc_files/local/proftpd.conf

```diff
@@ -0,0 +1,172 @@
+<%
+    import os
+
+    ftp = middleware.call_sync("ftp.config")
+    network_configuration = middleware.call_sync("network.configuration.config")
+%>
+
+ServerName "${network_configuration['hostname']} FTP Server"
+ServerType standalone
+DefaultServer on
+DefaultAddress localhost
+UseIPv6 on
+Port ${ftp['port']}
+User nobody
+Group nogroup
+Umask ${ftp['filemask']} ${ftp['dirmask']}
+SyslogFacility ftp
+MultilineRFC2228 off
+DisplayLogin /var/run/proftpd/proftpd.motd
+DeferWelcome off
+TimeoutIdle ${ftp['timeout']}
+TimeoutLogin 300
+TimeoutNoTransfer 300
+TimeoutStalled 3600
+MaxInstances none
+% if ftp['clients']:
+    MaxClients ${ftp['clients']}
+% endif
+% if ftp['ipconnections']:
+    MaxConnectionsPerHost ${ftp['ipconnections']}
+% endif
+% if ftp['loginattempt']:
+    MaxLoginAttempts ${ftp['loginattempt']}
+% endif
+DefaultTransferMode ascii
+AllowForeignAddress ${'on' if ftp['fxp'] else 'off'}
+% if ftp['masqaddress']:
+    MasqueradeAddress ${ftp['masqaddress']}
+% endif
+IdentLookups ${'on' if ftp['ident'] else 'off'}
+UseReverseDNS ${'on' if ftp['reversedns'] else 'off'}
+% if ftp['passiveportsmin']:
+    PassivePorts  ${ftp['passiveportsmin']} ${ftp['passiveportsmax']}
+% endif
+
+% if ftp['onlyanonymous']:
+    % if ftp['anonpath'] and os.path.isdir(ftp['anonpath']):
+        <Anonymous ${ftp['anonpath']}>
+            User ftp
+            Group ftp
+            UserAlias anonymous ftp
+            % if ftp['anonuserbw']:
+                TransferRate STOR ${ftp['anonuserbw']}
+            % endif
+            % if ftp['anonuserdlbw']:
+                TransferRate RETR ${ftp['anonuserdlbw']}
+            % endif
+            <Limit LOGIN>
+                AllowAll
+            </Limit>
+        </Anonymous>
+    % endif
+% elif ftp['onlylocal']:
+    <Limit LOGIN>
```

Comment:
Is the indentation intended here and above as well ? Looking at the ix.rc.d script, I don't think that's the case

Reply:
It is, agreed. That resolves this